### PR TITLE
Added wsk entity testing fixture to remove assets created during test

### PR DESCRIPTION
**Pull request by rabbah**
Added test fixture to sanitize entities before and after tests. This pattern avoids mistakenly sanitizing entities in the wrong collection and simplifies writing tests.

### CODE REVIEW

**Comment by rabbah** [comment]
playground build 42.

File: tests/src/system/basic/WskBasicTests.scala [hunk 1]
```diff
@@ -304,64 +251,76 @@ class WskBasicTests
...
-        val activationId = wsk.action.extractActivationId(activation)
-        activationId shouldBe a[Some[_]]
-
-        val expected = "ReferenceError" // representing nodejs giving an error when given malformed.js
-        val (found, logs) = wsk.activation.contains(activationId.get, expected)
-        assert(found, s"Did not find '$expected' in activation($activationId) ${logs getOrElse "empty"}")
+    it should "create and invoke action with malformed js resulting in activation error" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "MALFORMED"
+            assetHelper.withCleaner(wsk.action, name) {
```
Comment (psuter) [inline tests/src/system/basic/WskBasicTests.scala]: To me this looks like the action dies at the end of the scope, but it really dies at the end of the `withAssetCleaner` scope?

Comment (rabbah) [inline tests/src/system/basic/WskBasicTests.scala]:
```
 assetHelper.withCleaner(wsk.action, name) {
      (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("malformed.js")))
}
```

is creating the action and registering it for cleanup at the end; this itself should succeed (the action should be created).

**Comment by psuter** [comment]
Rebased and merged in: 7e2e38a119eb5cf45ffbeff0bd772e870ba42871 .

**Pull request closed without merging**